src/main/java/com/arpnetworking/metrics/mad/sources/MappingSource.java

```diff
@@ -116,47 +119,68 @@ public void notify(final Observable observable, final Object event) {
 
             // Merge the metrics in the record together
             final Record record = (Record) event;
-            final Map<String, MergingMetric> mergedMetrics = Maps.newHashMap();
+            final Map<ImmutableMap<String, String>, Map<String, MergingMetric>> mergedMetrics = Maps.newHashMap();
             for (final Map.Entry<String, ? extends Metric> metric : record.getMetrics().entrySet()) {
                 boolean found = false;
                 for (final Map.Entry<Pattern, List<String>> findAndReplace : _findAndReplace.entrySet()) {
                     final Matcher matcher = findAndReplace.getKey().matcher(metric.getKey());
                     if (matcher.find()) {
                         for (final String replacement : findAndReplace.getValue()) {
-                            merge(metric.getValue(), matcher.replaceAll(replacement), mergedMetrics);
+                            final String replacedString =
+                                    RegexAndMapReplacer.replaceAll(matcher, metric.getKey(), replacement, record.getDimensions());
+
+                            final int tagsStart = replacedString.indexOf(';');
```

Comment:
Extracting tags should be configurable based on the replacement.  For backwards compatibility default should probably be not to extract; but I imagine you would prefer the other.